File: backend/app/schemas/generation.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel

from app.schemas.research import ResearchReportDetail


TaskStatus = Literal["pending", "running", "succeeded", "failed"]


class GenerationTaskResponse(BaseModel):
    task_id: str
    symbol: str
    status: TaskStatus
    created_at: datetime
    finished_at: datetime | None = None
    error_message: str | None = None


class GenerationResultResponse(GenerationTaskResponse):
    result: ResearchReportDetail | None = None

File: frontend/src/types/generation.ts
import type { ResearchReportDetail } from './research'

export type GenerationStatus = 'pending' | 'running' | 'succeeded' | 'failed'

export interface GenerationTaskResponse {
  task_id: string
  symbol: string
  status: GenerationStatus
  created_at: string
  finished_at: string | null
  error_message: string | null
}

export interface GenerationResultResponse extends GenerationTaskResponse {
  result: ResearchReportDetail | null
}
